### DISCUSSION TITLE
use cache

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#200] [Ideas]
Hi, @wu-clan 

for example：get_location_online

[https://github.com/long2ice/fastapi-cache](https://github.com/long2ice/fastapi-cache)

**Comment by wu-clan** [comment]
We can indeed add data caching to speed up queries or responses, but I don't get what this library is for

**Reply by downdawn** [reply]
Mainly targeting infrequently changing data for caching, such as frequently accessed information, external API data, and time-consuming queries, etc. Manual implementation of caching is also feasible, but it might lack standardization and conciseness.

**Reply by wu-clan** [reply]
I've looked at the source code, and it's really just simple caching, but encrypted.

Its caching is reflected in the function return, but these things I would choose to implement manually in the business code, it will be more flexible, on the contrary, I don't know what is the advantage of this decorator

**Reply by downdawn** [reply]
Agreed. 

I've also looked at the source code of fastapi_limiter, and it's quite straightforward. However, if the business requirement involves rate limiting based on different roles for different users, it might become a bit challenging.

**Comment by downdawn** [comment]
In fact, the functionalities of these third-party libraries for FastAPI are still relatively simple; third-party libraries for Flask and Django are much more powerful.

[https://flask-caching.readthedocs.io/en/latest/](https://flask-caching.readthedocs.io/en/latest/)

[https://flask-limiter.readthedocs.io/en/stable/](https://flask-limiter.readthedocs.io/en/stable/)

**Reply by wu-clan** [reply]
fastapi is still young, we can't expect too much.
